tools/m2kcli/commands/analog/analog_in.cpp

```diff
@@ -0,0 +1,423 @@
+					  "                 [-q | --quiet]\n"
+					  "                 [-C | --calibrate]\n"
+					  "                 [-v | --voltage channel=<index>... raw=<value>]\n"
+					  "                 [-c | --capture channel=<index>... buffer_size=<size> raw=<value> [format=<type>]]\n"
+					  "                 [-g | --get <attribute> ...]\n"
+					  "                 [-G | --get-channel channel=<index> <attribute> ...]\n"
+					  "                 [-s | --set <attribute>=<value> ...]\n"
+					  "                 [-S | --set-channel channel=<index> <attribute>=<value> ...]\n"
+					  "\n"
+					  "Pinout:\n"
+					  "\u25A4 \u25A5 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1\n"
+					  "\u25A4 \u25A5 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1\n"
+					  "\n"
+					  "Positional arguments:\n"
+					  "  uri                   describe the context location \n"
+					  "                        auto | ip:192.168.2.1 | usb:XX.XX.X\n"
+					  "Optional arguments:\n"
+					  "  -h, --help            show this help message and exit\n"
+					  "  -q, --quiet           return result only\n"
+					  "  -v, --voltage channel=<index>... raw=<value>\n"
+					  "                        print the voltage of the given channel\n"
+					  "                        channel - {0 | 1}\n"
+					  "                        raw - 0 (processed values)\n"
+					  "                            - 1 (raw values)\n"
+					  "  -c, --capture channel=<index>... buffer_size=<size> raw=<value> [format=<type>]\n"
+					  "                        print a specific number of samples\n"
+					  "                        format - {csv | binary}; default csv\n"
+					  "  -g, --get [<attribute>...]\n"
+					  "                        return the value of the specified global attributes\n"
+					  "                        attribute:\n"
+					  "                            sampling_frequency\n"
+					  "                            oversampling_ratio\n"
+					  "                            trigger_source\n"
+					  "                            trigger_delay\n"
+					  "                            all\n"
+					  "  -G, --get-channel channel=<index> [<attribute> ...]\n"
+					  "                        return the value of the attributes corresponding to the given channel\n"
+					  "                        attributes:\n"
+					  "                            range\n"
+					  "                            trigger_level\n"
+					  "                            trigger_analog_condition\n"
+					  "                            trigger_mode\n"
+					  "                            trigger_hysteresis\n"
+					  "                            all\n"
+					  "  -s, --set [<attribute>=<value>...]\n"
+					  "                        set the value of the specified global attributes\n"
+					  "                        attribute:\n"
+					  "                            sampling_frequency - {10 | 100 | ... | 100000000}\n"
```

Comment:
The sampling_frequency attribute for m2k-adc goes from 1E3 to 1E8 .

Reply:
The correct available values were added in commit a61547e.